### PR TITLE
Feature/general hw

### PR DESCRIPTION
**Pull request by okurz**
Parts of https://github.com/os-autoinst/os-autoinst/pull/1898/commits for more small grained deployments

### CODE REVIEW

File: backend/generalhw.pm [hunk 1]
```diff
@@ -52,13 +53,18 @@ sub run_cmd ($self, @args) {
...
+        local $SIG{CHLD};
+        eval { $ret = IPC::Run::run([@full_cmd], \$stdin, \$stdout, \$stderr) };
+        die "Unable to run command '@full_cmd' (deduced from test variable $cmd): $@\n" if $@;
+    }
     chomp $stdout;
     chomp $stderr;
 
-    die "$args[0]: $stderr" unless $ret;
-    bmwqemu::diag("IPMI: $stdout");
+    die "$cmd: stdout: $stdout, stderr: $stderr" unless $ret;
```
Comment (marmarek) [inline backend/generalhw.pm:66]: `$cmd` is only introduced in https://github.com/os-autoinst/os-autoinst/pull/1898/commits/e2fc32e2b2743ada8e27e1b87f9d890f63e836fc

**Review by marmarek — CHANGES_REQUESTED** [review]

File: backend/generalhw.pm (hunk 1 above)
Comment (okurz) [reply backend/generalhw.pm]: ok, removed the logging change commit

**Comment by marmarek** [comment]
The https://github.com/os-autoinst/os-autoinst/pull/1918/commits/d43affa54fa370e2fc2fc7ecb172f7bb8c9c15c5 also uses `$cmd`, maybe better include https://github.com/os-autoinst/os-autoinst/commit/e2fc32e2b2743ada8e27e1b87f9d890f63e836fc too?

File: backend/generalhw.pm (hunk 1 above)
Comment (okurz) [reply backend/generalhw.pm:55]: added it back due to https://github.com/os-autoinst/os-autoinst/pull/1918#issuecomment-1013909982

**Comment by okurz** [comment]
> The [d43affa](https://github.com/os-autoinst/os-autoinst/commit/d43affa54fa370e2fc2fc7ecb172f7bb8c9c15c5) also uses `$cmd`, maybe better include [e2fc32e](https://github.com/os-autoinst/os-autoinst/commit/e2fc32e2b2743ada8e27e1b87f9d890f63e836fc) too?

Yes, I did that. This will cause one conflict regarding "reconnect_video_stream" but I am sure you would be able to handle that easily :)

**Comment by marmarek** [comment]
`./99-full-stack.t` failed, but I'm not sure why and if it's related to any of the changes here. Otherwise looks fine.

**Comment by Martchus** [comment]
I'm not reviewing that one again. Not sure whether splitting this into more PRs makes sense.

**Comment by okurz** [comment]
> `./99-full-stack.t` failed, but I'm not sure why and if it's related to any of the changes here. Otherwise looks fine.

It's not related to the changes here. I have seen this problem locally sporadically unrelated to these changes. I have reported this problem now in https://progress.opensuse.org/issues/104971 and retriggered the GHA CI job.

**Comment by okurz** [comment]
@Martchus included https://github.com/os-autoinst/os-autoinst/pull/1898#discussion_r785912622 and https://github.com/os-autoinst/os-autoinst/pull/1898#discussion_r785915979 here.

**Review by Martchus — APPROVED** [review]

**Comment by okurz** [comment]
Counting informal approval from marmarek as second one :)

**Pull request merged**